src/components/dashboards/Client-dashboard.tsx

```diff
@@ -1,245 +1,253 @@
+  const [selectedRequest, setSelectedRequest] = useState<any | null>(null);
+  const prevRequestsRef = useRef<any[]>([]);
+  const [isFetching, setIsFetching] = useState(false);
+  const [isModalClosing, setIsModalClosing] = useState(false);
+
+  const handleCloseModal = () => {
+    setIsModalClosing(true);
+    setTimeout(() => {
+      setSelectedRequest(null);
+      setIsModalClosing(false);
+    }, 300);
+  };
+
+  useEffect(() => {
+    if (status === "unauthenticated") {
+      router.replace("/authenticate");
+    }
+  }, [status, router]);
+
   useEffect(() => {
     if (session?.user?.isAdmin) {
       router.replace("/dashboard/admin");
+      return;
     }
-  }, [session?.user?.isAdmin, router]);
 
-  useEffect(() => {
-    // Fetch the residents requests
     if (session?.user?.id) {
       fetchResidentRequestsForUser();
     }
+  }, [session?.user]);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      if (session?.user?.id && !session?.user?.isAdmin) {
+        fetchResidentRequestsForUser(true);
+      }
+    }, 3000);
+    return () => clearInterval(interval);
   }, [session?.user?.id]);
 
-  const fetchResidentRequestsForUser = async () => {
+  const fetchResidentRequestsForUser = async (isPolling = false) => {
     try {
-      setIsLoading(true);
+      if (!session || !session.user?.id || session.user.isAdmin) {
+        return;
+      }
+
+      if (isFetching) return;
+      setIsFetching(true);
+
       const res =
         await ResidentRequestService.getResidentRequestsByAuthenticatedUser(
-          session!.user!.id!
+          session.user.id
         );
 
       const sortedRequests = res.requests.sort((a, b) => {
         return (
           new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime()
         );
       });
-      setRequestsForUser({ ...res, requests: sortedRequests });
-      setIsLoading(false);
+
+      if (prevRequestsRef.current) {
+        detectStatusChanges(prevRequestsRef.current, sortedRequests);
+      }
+
+      prevRequestsRef.current = sortedRequests;
+      setRequestsForUser(sortedRequests);
     } catch (error) {
-      console.error(error);
-      setApiError("Failed to fetch requests");
+      console.error("Error fetching resident requests:", error);
+      if (!isPolling) {
+        toast.error("Failed to fetch requests. Please try again.");
+      }
+    } finally {
+      setIsFetching(false);
       setIsLoading(false);
     }
   };
 
+  const detectStatusChanges = (prevRequests: any[], currentRequests: any[]) => {
+    currentRequests.forEach((currentRequest) => {
+      const prevRequest = prevRequests.find(
+        (req) => req.id === currentRequest.id
+      );
+
+      if (!prevRequest || prevRequest.status === currentRequest.status) {
+        return;
+      }
+
+      const addressInfo = `${currentRequest.address.city}, ${currentRequest.address.streetName}`;
+      if (currentRequest.status === "COMPLETED") {
```

Comment:
If we can find a better typing for the function we can get some better intellisense/ autocompletion - what happens if I spell it "completed"?

Reply:
there is no difference regarding to intellisense.
This is a common convention. It's easier to follow this is a fixed value and not a constant

Reply:
I tested this with "completed" and it doesn't highlight it as an error.